### DISCUSSION TITLE
Is it possible to push to a repository/edit a repository file via HTTP POST request?

### DISCUSSION THREAD

**Discussion by xTerrene** [community/community#41904] [Other Feature Feedback, Questions, & Ideas]
Hopefully this is the right place to ask. A little upset they killed off the email service, but I digress.

I'm aware that, as it currently is, repository contributors or owners can get raw file contents via HTTP request to the raw githubusercontent domain leading to them.

I'm curious if, despite potential security concerns, it's possible to _push_ to a repository via HTTP request, as well? For example, if I want to have a file in the repository get updated through curl, could I send a POST request with my token in the Authorization header?

Part of me doubts this in consideration of potential security problems depending on who puts what where and how, but I figured it wouldn't hurt to ask.

**Comment by truh** [comment]
Git on Github works via SSH and HTTPS. The git http-protocol is described here https://www.git-scm.com/docs/http-protocol.

**Comment by airtower-luna** [comment]
The REST API provides that option, see [Create or update file contents](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28#create-or-update-file-contents). Mind that when updating a file you have to provide the blob ID of the previous version (see the `sha` parameter under "Body parameters").
